### ISSUE TITLE
When updating the "path" property of firebase-query element, the template dom-repeat displays no items

### ISSUE THREAD

**Issue by benj13260** [firebase/polymerfire#100]
### Description

When updating the "path" property of firebase-query element, the template dom-repeat displays no items.

My new created element is receiving an entry parameter (chefID), and based on this information the path of the query is updated to retrieve the required data (list of recipe)

Initially this was done in the firebase-query element

```
    <firebase-query id="query"  app-name="app" log data="{{data}}" 
        path= "chefs/[[chefID]]/recipes" >
    </firebase-query>

     <template i is="dom-repeat" items="[[data]]">
```

On the first load, the repeat template was showing correct result.
When a new path was loaded, the repeat template was empty (whatever was retrieved by the query).

Looking at the log provided by the firebase-query element, it looked that the information was correctly retrieved; but something was going wrong with the dom-repeat element.

**I realised that before assigning a new value to the "path" it should first be set to null.**

Therefore the path is now updated in a javascript function and not by binding:

```
     this.$.query.path = null;
     this.$.query.path = 'chefs/' + this.chefID + '/receipes';
```

By setting the path to null. the dom-repeat element is "cleaned" somehow and new values can then flow-in

I don't know if this is a bug (on firebase-query element or on dom-repeat element), 
It might be interesting to update polymerfire documentation to mention this trick.

**Comment by bvhme** [comment]
I am having similar issues, though I have the problem that it will first get data for `chefs/recipes` and then when the `[[chefID]]` is populated doesn't get data for the new array. E.g. if I set path to `/posts/[[postId]]` I get the value for `/posts/` and I don't get the value for the right element.

**Comment by isaiahnavasquez** [comment]
You are a hero @benj13260 ! Your alternate solution just help me with my project. Thanks!

**Comment by brycethorup** [comment]
I would also like to recommend @benj13260's solution. I just got done spending 5 hours troubleshooting this. Is this expected behavior, or a bug? I noticed that changing the `path` attribute using data-bound variables causes the query to be requested twice, first with the correct array of values and second with an empty array.

**Comment by dam0vm3nt** [comment]
Hello, I'm having the same problem and I realized that when the path changes the second time there are two places where the data gets cleared :

 - https://github.com/firebase/polymerfire/blob/master/firebase-query.html#L270 
 - https://github.com/firebase/polymerfire/blob/master/firebase-query.html#L270

So @benj13260 workaround basically works because it clears the previous query but obviously there's something wrong that needs to be fixed.

**Comment by santanafabio** [comment]
Same here! Anyone knows what to do?

**Comment by christophe-g** [comment]
Yes, this is a pretty serious issue - see a proposed fix above.

**Comment by hariprasadiit** [comment]
facing the same problem. If we update `firebase-document` path property dynamically, it doesn't refreshes the data

**Comment by christophe-g** [comment]
@hariprasadiit 
Nope, #95 didn't solve the issue  for me. 
Take a look at https://github.com/firebase/polymerfire/pull/167 for a fix

**Comment by merlinnot** [comment]
@mbleigh @cdata Could you please take a look at this issue and proposed fix #167? It's a serious issue which can be easily fixed.

**Comment by MarcMouallem** [comment]
Fix #167 looks good to me, I am doing the same thing locally. Let's do this thing!
